pallets/mnt-token/src/lib.rs

```diff
@@ -202,19 +520,36 @@ impl<T: Config> Pallet<T> {
 	fn refresh_mnt_speeds() -> DispatchResult {
 		// TODO Add update indexes here when it will be implemented
 		let (pool_utilities, sum_of_all_utilities) = Self::calculate_enabled_pools_utilities()?;
-		let sum_of_all_utilities = Rate::from_inner(sum_of_all_utilities);
-		if sum_of_all_utilities == Rate::zero() {
+		if sum_of_all_utilities == Balance::zero() {
 			// There is nothing to calculate.
 			return Ok(());
 		}
 		let mnt_rate = Self::mnt_rate();
 		for (currency_id, utility) in pool_utilities {
-			let utility = Rate::from_inner(utility);
-			let utility_fraction = utility
-				.checked_div(&sum_of_all_utilities)
+			let utility_fraction = Rate::saturating_from_rational(utility, sum_of_all_utilities);
+			let pool_mnt_speed = Rate::from_inner(mnt_rate)
+				.checked_mul(&utility_fraction)
 				.ok_or(Error::<T>::NumOverflow)?;
-			let pool_mnt_speed = mnt_rate.checked_mul(&utility_fraction).ok_or(Error::<T>::NumOverflow)?;
-			MntSpeeds::<T>::insert(currency_id, pool_mnt_speed);
+			MntSpeeds::<T>::insert(currency_id, pool_mnt_speed.into_inner());
+		}
+		Ok(())
+	}
+
+	/// Transfer mnt tokens to user balance if it possible. Otherwise, put them into internal
+	/// storage.
+	///
+	/// - `user`: MNT tokens recipient.
+	/// - `user_accrued`: The total amount of accrued tokens.
+	/// - `threshold`: The threshold above which the flywheel transfers MNT.
+	fn transfer_mnt(user: &T::AccountId, user_accrued: Balance, threshold: Balance) -> DispatchResult {
+		if user_accrued >= threshold && user_accrued > 0 {
+			// TODO check is currency in MNT pallet enough.
+			// Need to discuss what we should do.
+			// Erorr/Event/save money to MntAccrued/stop producing mnt tokens
+			T::MultiCurrency::transfer(MNT, &Self::get_account_id(), &user, user_accrued)?;
+			MntAccrued::<T>::remove(user); // set to 0
+		} else {
+			MntAccrued::<T>::insert(user, user_accrued);
 		}
 		Ok(())
 	}
```

Comment:
In my opinion, this function should be renamed to update_accrued and accept **accrue_delta** instead of **user_accrued**.
```suggestion
	fn update_accrued(user: &T::AccountId, accrue_delta: Balance, threshold: Balance) -> DispatchResult {
		let mut user_accrued = MntAccrued::<T>::get(user)
			.checked_add(accrue_delta.into_inner())
			.ok_or(Error::<T>::NumOverflow)?;
		if user_accrued >= threshold && user_accrued > 0 {
			// TODO check is currency in MNT pallet enough.
			// Need to discuss what we should do.
			// Erorr/Event/save money to MntAccrued/stop producing mnt tokens
			T::MultiCurrency::transfer(MNT, &Self::get_account_id(), &user, user_accrued)?;
			MntAccrued::<T>::remove(user); // set to 0
		} else {
			MntAccrued::<T>::insert(user, user_accrued);
		}
		Ok(())
	}
```

Reply:
I think it is not a good idea, Because it is not a delta
Se comment to code: `user_accrued`: The total amount of accrued tokens.

Reply:
It is made by analogy with:

```
    function transferComp(address user, uint userAccrued, uint threshold) internal returns (uint) {
        if (userAccrued >= threshold && userAccrued > 0) {
            Comp comp = Comp(getCompAddress());
            uint compRemaining = comp.balanceOf(address(this));
            if (userAccrued <= compRemaining) {
                comp.transfer(user, userAccrued);
                return 0;
            }
        }
        return userAccrued;
    }
```